projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -312,41 +317,47 @@ export class TableCdkDataSource implements DataSource<TableRow> {
       return of([]);
     }
 
+    let total = 0;
+
     return this.tableDataSourceProvider.data.getData(request).pipe(
       tap(response => this.updatePaginationTotalCount(response.totalCount)),
-      map(response => response.data),
+      map(response => {
+        total = response.totalCount;
```

Comment:
nit: we can do it inside the L323's `tap`?

Reply:
Yeah, no problem. I can change in my local and let it ride in the next one. Thanks!